Framework/HistogramData/src/Histogram.cpp

```diff
@@ -208,6 +254,20 @@ void Histogram::setUncertainties(const FrequencyStandardDeviations &e) {
   setFrequencyStandardDeviations(e);
 }
 
+void Histogram::checkAndSetYModeCounts() {
+  if (yMode() == YMode::Frequencies)
+    throw std::logic_error("Histogram: Y is storing Counts, modifying "
+                           "Frequencies is not possible.");
+  m_yMode = YMode::Counts;
+}
```

Comment:
So I think this is going to allow you to set the `YMode` to counts if uninitialised as well as if already `YMode::Counts`, which I think is correct.